### PR TITLE
Decorate `reduced_mass` with `particle_input` & `validate_quantities` whilst expanding its docstring

### PR DESCRIPTION
**Pull request by namurphy**
- Decorated `plasmapy.particles.atomic.reduced_mass` with `@particle_input` and `@validate_quantities`
   - Removed a function inside `reduced_mass` that extracted the particle mass
 - Expanded the docstring of `reduced_mass` to include its equation and some properties

### CODE REVIEW

**Comment by namurphy** [comment]
Coincidentally, I'm wondering about moving `reduced_mass` to a separate module in `plasmapy.particles`, since `atomic.py` might not be the best fit.  If we address #1920, we could possibly move `reduced_mass` into a module with the density functions, like maybe `plasmapy.particles.utilities`.  But, that's not for this issue.

File: plasmapy/particles/atomic.py [hunk 1]
```diff
@@ -833,46 +844,27 @@ def reduced_mass(test_particle, target_particle) -> u.Quantity:
...
-                f"Unable to find the reduced mass because the mass of "
-                f"{particle} is not available."
-            ) from None
-        else:
-            return particle.mass.to(u.kg)
-
-    test_mass = get_particle_mass(test_particle)
-    target_mass = get_particle_mass(target_particle)
-
-    return (test_mass * target_mass) / (test_mass + target_mass)
```
Comment (namurphy) [inline plasmapy/particles/atomic.py:875]: Feels nice to be able to delete this much code!

**Comment by namurphy** [comment]
This is pretty close, but this might be a good time to improve the test coverage of `reduced_mass` a bit.  It would be good to merge #1884 before this one, so that I can add some additional test cases that #1884 would enable.

**Review by StanczakDominik — APPROVED** [review]
Skimmed, looks great for now 👍🏻

**Pull request merged**